### PR TITLE
Prevent HTML hightlight code from being escaped on entity results vie…

### LINKED ISSUE #461: Results are displaying HTML added by Transvision when searching for entities
When we are searching for entity name (search_type=entities), the HTML code used to display a fake non-breaking space (<span class="highlight-gray"> </span>) is displayed in the page, like in this one:

https://transvision.mozfr.org/?sourcelocale=en-US&locale=fr&repo=central&search_type=entities&recherche=suite/chrome/common/notification.properties:vulnerableNoUpdatePluginWarning

The same string looks fine while searching the string itself: https://transvision.mozfr.org/?recherche=Vulnerable+Plugin!&repo=central&sourcelocale=en-US&locale=fr&search_type=strings

I think it can be reproduced with any string containing thin or regular nbsp.
"search_type=strings_entities" does not seem affected

### PR DESCRIPTION
**Pull request by TheoChevalier**
…w - fixes #461

### CODE REVIEW

**Comment by TheoChevalier** [comment]
Crossing my fingers that this is not causing more regressions than that it's fixing, this code area looks… sensitive :)

**Comment by TheoChevalier** [comment]
I knew I'd forgot something… When filing a bug from entity view, the string contains HTML code.

File: app/views/results_entities.php [hunk 1]
```diff
@@ -23,12 +23,18 @@
         $path_locale2 = VersionControl::hgPath($locale, $check['repo'], $entity);
     }
 
-    $target_string2 = '';
+    // Escape strings for HTML Display
```
Comment (flodolo) [inline app/views/results_entities.php]: display (lowercase)

File: app/views/results_entities.php [hunk 2]
```diff
@@ -23,12 +23,18 @@
...
+    $bz_target_string = $target_string = isset($tmx_target[$entity])
+                                            ? Utils::secureText($tmx_target[$entity])
+                                            : '';
+    $bz_target_string2 = $target_string2 = isset($tmx_target2[$entity])
```
Comment (flodolo) [inline app/views/results_entities.php]: You should put this definition in the following if cycle, we don't need it for the normal view, only for 3locales.

File: app/views/results_entities.php [hunk 3]
```diff
@@ -23,12 +23,18 @@
...
+                                            ? Utils::secureText($tmx_target2[$entity])
+                                            : '';
+    $source_string = Utils::secureText($tmx_source[$entity]);
+
     if ($url['path'] == '3locales') {
-        if (isset($tmx_target2[$entity])) {
-            // nbsp highlight
-            $target_string2 = str_replace(' ', '<span class="highlight-gray"> </span>', $tmx_target2[$entity]);
-        }
+        // nbsp highlight only after strings have been escaped
```
Comment (flodolo) [inline app/views/results_entities.php]: ```// Highlight non-breaking spaces``

File: app/views/results_entities.php [hunk 4]
```diff
@@ -37,17 +43,8 @@
...
-    } else {
-        $target_string = '';
-    }
-
-    // Escape strings for HTML Display
-    $source_string  = Utils::secureText($tmx_source[$entity]);
-    $target_string  = Utils::secureText($target_string);
-    $target_string2 = Utils::secureText($target_string2);
+    // nbsp highlight only after strings have been escaped
+    $target_string = str_replace(' ', '<span class="highlight-gray"> </span>', $target_string);
```
Comment (flodolo) [inline app/views/results_entities.php]: Same note about the comment, I would move this one to line 30 where you define `$bz_target_string = $target_string`

**Comment by TheoChevalier** [comment]
Thanks :) Should be fixed

**Pull request merged**